cli/src/main/scala/com/lightbend/rp/reactivecli/runtime/kubernetes/Service.scala

```diff
@@ -51,29 +51,33 @@ object Service {
   /**
    * Generates the [[Service]] resource.
    */
-  def generate(annotations: Annotations, apiVersion: String, clusterIp: Option[String]): Try[Service] =
-    annotations.appName match {
-      case Some(appName) =>
-        Success(
-          Service(
-            appName,
-            Json(
-              "apiVersion" -> apiVersion.asJson,
-              "kind" -> "Service".asJson,
-              "metadata" -> Json(
-                "labels" -> Json(
-                  "app" -> annotations.appName.asJson),
-                "name" -> annotations.appName.asJson)
-                .deepmerge(
-                  annotations.namespace.fold(jEmptyObject)(ns => Json("namespace" -> ns.asJson))),
-              "spec" -> Json(
-                "clusterIP" -> clusterIp.getOrElse("None").asJson,
-                "ports" -> annotations.endpoints.asJson,
-                "selector" -> Json(
-                  "app" -> annotations.appName.asJson)))))
-      case _ =>
-        Failure(new IllegalArgumentException("Unable to generate Kubernetes service resource: application name is required"))
-    }
+  def generate(annotations: Annotations, apiVersion: String, clusterIp: Option[String]): Try[Option[Service]] =
```

Comment:
Type change: `Service` should be optional -- an app doesn't have to define endpoints.